aws/data_source_aws_elb_test.go

```diff
@@ -0,0 +1,113 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAWSELB_basic(t *testing.T) {
+	lbName := fmt.Sprintf("testaccawselb-basic-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAWSELBConfigBasic(lbName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "name", lbName),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "cross_zone_load_balancing", "true"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "idle_timeout", "30"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "internal", "true"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "subnets.#", "2"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "security_groups.#", "1"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "tags.%", "1"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "tags.TestName", "TestAccAWSELB_basic"),
+					resource.TestCheckResourceAttrSet("data.aws_elb.elb_test", "dns_name"),
+					resource.TestCheckResourceAttrSet("data.aws_elb.elb_test", "zone_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAWSELBConfigBasic(lbName string) string {
+	return fmt.Sprintf(`resource "aws_elb" "elb_test" {
+  name            = "%s"
+  internal        = true
+  security_groups = ["${aws_security_group.elb_test.id}"]
+  subnets         = ["${aws_subnet.elb_test.*.id}"]
+
+  idle_timeout = 30
+  enable_deletion_protection = false
+
+  listener {
+    instance_port     = 80
+    instance_protocol = "http"
+    lb_port           = 80
+    lb_protocol       = "http"
+  }
+
+  tags {
+    TestName = "TestAccAWSELB_basic"
```

Comment:
Nitpick: it may be useful to make these tags in the test config data source specific.

Reply:
`t.Name()` to the rescue!

<img width="811" alt="screenshot 2017-12-06 19 31 22" src="https://user-images.githubusercontent.com/189114/33692617-1c6c854e-dabc-11e7-83a2-bdf3607cb063.png">